app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -94,10 +105,33 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
+      try {
+        if (result.state.interaction.id.equals("NumericInput"))
+          addNumberInputContentCard(result.state.interaction.customizationArgsMap.get("placeholder")!!.normalizedString)
+      } catch (e: Exception) {
+      }
       logger.d("StateFragment", "getCurrentState: ${result.state.name}")
     })
   }
 
+  /** The function for adding [NumberInputInteractionView]. */
+  fun addNumberInputContentCard(placeholder: String) {
+    var contentComponent = NumberInputInteractionView(
+      fragment.context!!,
+      placeholder,
+      1
+    )
+    val params = LinearLayout.LayoutParams(
+      LinearLayout.LayoutParams.MATCH_PARENT,
+      LinearLayout.LayoutParams.WRAP_CONTENT
+    )
+    params.setMargins(dpToPx(8), dpToPx(8), dpToPx(8), dpToPx(8))
+    llRoot!!.addView(contentComponent, params)
```

Comment:
Once you implement the `lateinit` comment mentioned earlier.

you can update this code to  
`llRoot.addView(contentComponent, params)`

Ditto for other views.

Reply:
resolved